VisualizationBase/src/items/Item.cpp

```diff
@@ -419,6 +421,10 @@ void Item::removeFromScene()
 		// Mark this item as not needing updates
 		scene()->setItemIsSensitiveToScale(this, false);
 
+		//Notify all views about deletion of the item
+		for (auto view : scene()->viewItems()->viewItemsAsList())
```

Comment:
Instead of manually iterating over all the views, just call a `notify` method on the `viewManager`. This way you let the manager decide what to do, which is the correct place to the iteration. This will also remove the need to include the `ViewItem.h` file above.